core/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```diff
@@ -98,28 +98,24 @@ public RegexpQueryBuilder queryName(String queryName) {
     @Override
     public void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(RegexpQueryParser.NAME);
-        if (boost == -1 && rewrite == null && queryName != null) {
-            builder.field(name, regexp);
```

Comment:
short version is gone in the builder, it was too error prone as the initial if was outdated.  _name is still supported in the short version by the parser, but the builder will always use the long version.